### PR TITLE
ROR Capstone Project

### PR DESCRIPTION
**Pull request by rolandfbk**
My Pocket is a Ruby on Rails application that allows the user to:

- register and log in, so that the data is private to them.
- create a category and create transactions for that category.
- see the money spent on each category.

I also created unit and integration tests for all the most important components of the app.

**[Link to the video presentation](https://www.loom.com/share/3b470b92251547e09c0fec99a3c446e2)**

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  def show
+    @category = Category.find(params[:id])
+    @payments = @category.payments.order(created_at: :desc).all
+    @total_payments = @category.total_payments
+  end
+
+  def category_params
+    params.require(:category).permit(:name, :icon)
+  end
```
Comment (Deepakdanger) [inline app/controllers/categories_controller.rb:30]: - [ ] Kindly use this method under `private`.

File: app/controllers/payments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    if @payment.save
+      redirect_to category_path(params[:category_id])
+    else
+      render :new
+    end
+  end
+
+  def payment_params
+    params.require(:payment).permit(:name, :amount, :category_id)
+  end
```
Comment (Deepakdanger) [inline app/controllers/payments_controller.rb:21]: - [ ] Kindly use this method under `private`.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @rolandfbk , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 
- Vido presentation is good :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by muneebulrehman — APPROVED** [review]
Hi @rolandfbk , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- Good documentation.
- No linter errors
- Good follow-up with the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**